server/src/main/protobuf/server.proto

```diff
@@ -33,4 +34,9 @@ service FileManagerService {
    to download the file.
    */
   rpc File (file.FileRequest) returns (file.FileResponse);
+
+  /*
+  Route called by clients when they ended their file upload.
+   */
+  rpc OnUploadEvent(upload.UploadEvent) returns (google.protobuf.Empty);
```

Comment:
You could have just use the previous route handling events for this and create an eventType "UploadACK"